Error compiling for Ubuntu

ERRO: **error compiling for Ubuntu**


/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:39:31: error: ‘siginfo_t’ has not been declared
 void crashHandler(int signum, siginfo_t* info, void* secret)
                               ^~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp: In function ‘void crashHandler(int, int*, void*)’:
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:12: error: ‘SIGILL’ was not declared in this scope
     signal(SIGILL, SIG_DFL);
            ^~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:12: note: suggested alternative: ‘CSIGNAL’
     signal(SIGILL, SIG_DFL);
            ^~~~~~
            CSIGNAL
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:20: error: ‘SIG_DFL’ was not declared in this scope
     signal(SIGILL, SIG_DFL);
                    ^~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:20: note: suggested alternative: ‘REG_EFL’
     signal(SIGILL, SIG_DFL);
                    ^~~~~~~
                    REG_EFL
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:5: error: ‘signal’ was not declared in this scope
     signal(SIGILL, SIG_DFL);
     ^~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:118:5: note: suggested alternative: ‘signum’
     signal(SIGILL, SIG_DFL);
     ^~~~~~
     signum
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:119:12: error: ‘SIGSEGV’ was not declared in this scope
     signal(SIGSEGV, SIG_DFL);
            ^~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:120:12: error: ‘SIGFPE’ was not declared in this scope
     signal(SIGFPE, SIG_DFL);
            ^~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:121:12: error: ‘SIGABRT’ was not declared in this scope
     signal(SIGABRT, SIG_DFL);
            ^~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp: In function ‘void installCrashHandler()’:
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:126:22: error: aggregate ‘installCrashHandler()::sigaction sa’ has incomplete type and cannot be defined
     struct sigaction sa;
                      ^~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:128:5: error: ‘sigemptyset’ was not declared in this scope
     sigemptyset (&sa.sa_mask);
     ^~~~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:129:19: error: ‘SA_RESTART’ was not declared in this scope
     sa.sa_flags = SA_RESTART | SA_SIGINFO;
                   ^~~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:129:19: note: suggested alternative: ‘ERESTART’
     sa.sa_flags = SA_RESTART | SA_SIGINFO;
                   ^~~~~~~~~~
                   ERESTART
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:129:32: error: ‘SA_SIGINFO’ was not declared in this scope
     sa.sa_flags = SA_RESTART | SA_SIGINFO;
                                ^~~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:131:15: error: ‘SIGILL’ was not declared in this scope
     sigaction(SIGILL, &sa, NULL);   // illegal instruction
               ^~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:131:15: note: suggested alternative: ‘CSIGNAL’
     sigaction(SIGILL, &sa, NULL);   // illegal instruction
               ^~~~~~
               CSIGNAL
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:131:32: error: invalid use of incomplete type ‘struct installCrashHandler()::sigaction’
     sigaction(SIGILL, &sa, NULL);   // illegal instruction
                                ^
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:126:12: note: forward declaration of ‘struct installCrashHandler()::sigaction’
     struct sigaction sa;
            ^~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:132:15: error: ‘SIGSEGV’ was not declared in this scope
     sigaction(SIGSEGV, &sa, NULL);  // segmentation fault
               ^~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:132:33: error: invalid use of incomplete type ‘struct installCrashHandler()::sigaction’
     sigaction(SIGSEGV, &sa, NULL);  // segmentation fault
                                 ^
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:126:12: note: forward declaration of ‘struct installCrashHandler()::sigaction’
     struct sigaction sa;
            ^~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:133:15: error: ‘SIGFPE’ was not declared in this scope
     sigaction(SIGFPE, &sa, NULL);   // floating-point exception
               ^~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:133:32: error: invalid use of incomplete type ‘struct installCrashHandler()::sigaction’
     sigaction(SIGFPE, &sa, NULL);   // floating-point exception
                                ^
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:126:12: note: forward declaration of ‘struct installCrashHandler()::sigaction’
     struct sigaction sa;
            ^~~~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:134:15: error: ‘SIGABRT’ was not declared in this scope
     sigaction(SIGABRT, &sa, NULL);  // process aborted (asserts)
               ^~~~~~~
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:134:33: error: invalid use of incomplete type ‘struct installCrashHandler()::sigaction’
     sigaction(SIGABRT, &sa, NULL);  // process aborted (asserts)
                                 ^
/home/anderson/Documentos/otclient-master/src/framework/platform/unixcrashhandler.cpp:126:12: note: forward declaration of ‘struct installCrashHandler()::sigaction’
     struct sigaction sa;

/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp: In static member function ‘static void TiXmlBase::ConvertUTF32ToUTF8(long unsigned int, char*, int*)’:
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:113:19: warning: this statement may fall through [-Wimplicit-fallthrough=]
             input >>= 6;
             ~~~~~~^~~~~
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:114:9: note: here
         case 3:
         ^~~~
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:117:19: warning: this statement may fall through [-Wimplicit-fallthrough=]
             input >>= 6;
             ~~~~~~^~~~~
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:118:9: note: here
         case 2:
         ^~~~
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:121:19: warning: this statement may fall through [-Wimplicit-fallthrough=]
             input >>= 6;
             ~~~~~~^~~~~
/home/anderson/Documentos/otclient/src/framework/xml/tinyxmlparser.cpp:122:9: note: here
         case 1:
         ^~~~

/home/anderson/Documentos/otclient/src/client/thingtype.cpp: In member function ‘void ThingType::unserialize(uint16, ThingCategory, const FileStreamPtr&)’:
/home/anderson/Documentos/otclient/src/client/thingtype.cpp:278:15: warning: variable ‘frameGroupType’ set but not used [-Wunused-but-set-variable]
         uint8 frameGroupType = FrameGroupDefault;
               ^~~~~~~~~~~~~~

/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp: In member function ‘void ProtocolGame::parseStore(const InputMessagePtr&)’:
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp:502:13: warning: variable ‘highlightState’ set but not used [-Wunused-but-set-variable]
         int highlightState = 0;
             ^~~~~~~~~~~~~~
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp: In member function ‘void ProtocolGame::parseCoinBalance(const InputMessagePtr&)’:
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp:522:9: warning: variable ‘coins’ set but not used [-Wunused-but-set-variable]
     int coins = -1;
         ^~~~~
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp:523:9: warning: variable ‘transferableCoins’ set but not used [-Wunused-but-set-variable]
     int transferableCoins = -1;
         ^~~~~~~~~~~~~~~~~
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp: In member function ‘void ProtocolGame::parseStoreTransactionHistory(const InputMessagePtr&)’:
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp:555:9: warning: variable ‘currentPage’ set but not used [-Wunused-but-set-variable]
     int currentPage;
         ^~~~~~~~~~~
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp: In member function ‘void ProtocolGame::parsePlayerStats(const InputMessagePtr&)’:
/home/anderson/Documentos/otclient/src/client/protocolgameparse.cpp:1354:7: warning: suggest explicit braces to avoid ambiguous ‘else’ [-Wdangling-else]
     if(g_game.getFeature(Otc::GameExperienceBonus))
       ^

[ 77%] Linking CXX executable otclient
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libogg.a(framing.o): relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompilar com -fPIC
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libogg.a(bitwise.o): relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompilar com -fPIC
/usr/bin/ld: falha no link final: Seção não representável na saída
collect2: error: ld returned 1 exit status
CMakeFiles/otclient.dir/build.make:3857: recipe for target 'otclient' failed
make[2]: *** [otclient] Error 1
CMakeFiles/Makefile2:99: recipe for target 'CMakeFiles/otclient.dir/all' failed
make[1]: *** [CMakeFiles/otclient.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

--- a/src/framework/platform/unixcrashhandler.cpp
+++ b/src/framework/platform/unixcrashhandler.cpp
@@ -32,6 +32,7 @@
 
 #include <execinfo.h>
 #include <ucontext.h>
+#include <signal.h>
 
 #define MAX_BACKTRACE_DEPTH 128
 #define DEMANGLE_BACKTRACE_SYMBOLS

